### ISSUE TITLE
tusimple training stuck at epoch #1

### ISSUE THREAD

**Issue by lilyzhng** [harryhan618/SCNN_Pytorch#32]
Hi Harry,

I have checked all the existed issues, and found no solution for this problem. The issue is that when launching tusimple training with the train.py, after train epoch #0, val epoch #0, it will always stuck at train epoch #1. And I wonder the results you get from the readme.md is also from only 1 epoch?

The following is the tensorboard, and you can see there is no logs from val loss. Is it normal?
![Screenshot from 2020-03-04 11-07-44](https://user-images.githubusercontent.com/10191088/75914049-ec929480-5e08-11ea-9f86-3cf8ad433e3a.png)

**Comment by lilyzhng** [comment]
After setting num_workers=0, this issue of hanging after epoch 1 gets resolved. But there is still no plot showing for val_loss until several epochs later.

@harryhan618 Could you help explain why would you put `tensorboard.scalar_summary("val_loss", val_loss, iter_idx)` outside of the for loop for line 184 in train.py?
<img width="850" alt="Screen Shot 2020-03-27 at 6 36 14 PM" src="https://user-images.githubusercontent.com/10191088/77811771-f00cea80-7059-11ea-8f20-d7b9a7c3af0e.png">

**Comment by harryhan618** [comment]
@alchemz hi！
I put tensorboard logging outside of the loop during validation, because during validation, I just want to know the loss situation for the whole epoch. Single iteration during validation is meaningless.
